user/super/com/google/gwt/emul/java/util/function/Predicate.java

```diff
@@ -47,4 +47,10 @@ default Predicate<T> or(Predicate<? super T> other) {
     checkCriticalNotNull(other);
     return t -> test(t) || other.test(t);
   }
+
+  @SuppressWarnings("unchecked")
+  static <T> Predicate<T> not(Predicate<? super T> other) {
+    checkCriticalNotNull(other);
```

Comment:
I don't think a critical check is needed here because `other` is called right away and would produce a NPE anyways. I think we don't need a check here at all, what do you think @niloc132 ?

Reply:
I think you're right - my initial thinking here was for the weird cases where the compiler decides that null is actually okay since the simple method call gets inlined away and the `this` might never be accessed, but in retrospect I'm not sure that's a thing here. NPEs tend to show up as JS TypeError  (there is some wiring in GWT that should transform that, but it doesn't seem to work all of the time), so a plain `checkNotNull` might be better than skipping it entirely?

To be more specific, `negate()` will almost always be transformed into
```java
public static <T> Predicate<T> negate(Predicate<T> this$static) {
    return t -> !this$static.test(t);
}
```

This could then be inlined, even if `this$static` is null, but wouldn't be identified until the lambda is actually invoked. From tracing `MakeCallsStatic.execJProgram program, boolean addRuntimeChecks, OptimizerContext optimizerCtx)`, the `addRuntimeChecks` value is usually false for production code - so the compiler doesn't synthesize calls to `Exceptions.checkNotNull(this$static)` when this transformation is done. 

We're no worse off than we were with just plain `willBeNull.negate()` though, so yeah I think we could loosen this.